ArticleTemplates/assets/js/modules/cards.js

```diff
@@ -1,96 +1,47 @@
 define([
-    'flipSnap',
     'modules/util'
-],
+], 
 function (
-    flipSnap,
     util
 ) {
     'use strict';
 
     var initialised = false;
 
-    function setupGlobals() {
-        window.articleCardsInserter = articleCardsInserter;
-        window.articleCardsFailed = articleCardsFailed;
-
-        window.applyNativeFunctionCall('articleCardsInserter');
-        window.applyNativeFunctionCall('articleCardsFailed');
-    }
-
-    function articleCardsInserter(html) {
-        var relatedContentWrapper = document.getElementsByClassName('related-content__wrapper')[0];
-
-        if (relatedContentWrapper) {
-            if (!html) {
-                articleCardsFailed();
-            } else {
-                relatedContentWrapper.innerHTML = html;
-                snapToGrid();
-            }
+    function ready(config) {
+        if (!initialised && GU.opts.platform === 'ios') {
+            initialised = true;
+            new window.MutationObserver(function(mutations) {
```

Comment:
I was thinking we could assign the last result of `getRelatedContentPosition()` sent to the native layer top a variable, then in the `function(mutations){ ... }` after we call `getRelatedContentPosition()` we can compare with the last result stored in memory, only if there's a difference do pass the result to `window.webkit.messageHandlers.bodyMutationNotification.postMessage`. This could save us sending position data to the native layer even if the position of the related content container hasn't changed. What do you think? Would it save the native layer continuously repositioning the native related cards?